lib/network/api/node_info.go

```diff
@@ -0,0 +1,40 @@
+package api
+
+import (
+	"net/http"
+
+	"boscoin.io/sebak/lib/common"
+	"boscoin.io/sebak/lib/node"
+)
+
+func (api NetworkHandlerAPI) GetNodeInfoHandler(w http.ResponseWriter, r *http.Request) {
+	nodeInfo := &node.NodeInfo{}
+	*nodeInfo = *&api.nodeInfo
+
+	nodeInfo.Node.State = api.localNode.State()
+
+	if nodeInfo.Node.Endpoint == nil {
+		rUrl := common.RequestURLFromRequest(r)
+		rUrl.Path = ""
+		rUrl.RawQuery = ""
+
+		nodeInfo.Node.Endpoint = common.NewEndpointFromURL(rUrl)
+	}
+
+	if api.GetLatestBlock != nil {
+		nodeInfo.Block = node.NodeBlockInfo{
+			Height:   api.GetLatestBlock().Height,
+			Hash:     api.GetLatestBlock().Hash,
+			TotalTxs: api.GetLatestBlock().TotalTxs,
```

Comment:
Since `GetLatestBlock` might involve IO, the function should be invoked once and the result used.